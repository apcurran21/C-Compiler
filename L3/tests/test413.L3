define @main() {
	%x <- 3
	%y <- 3
	%result1 <- call @findGCD(%x, %y)

	%x2 <- 30
	%y2 <- 50

	%result2 <- call @findGCD(%x2, %y2)

	%x3 <- 75
	%y3 <- 80
	%result3 <- call @findGCD(%x3, %y3)
	
	return
}

define @findGCD(%a, %b) {
	:TestNumbers
	%isLesser <- %a < %b
	br %isLesser :Lesser
	%isEqual <- %a = %b
	br %isEqual :Equal

	:Greater
	%a <- %a - %b
	call @printResult(%a, %b)
	call @findGCD(%a, %b)
	br :Finished

	:Lesser
	%b <- %b - %a
	call @printResult(%a, %b)
	call @findGCD(%a, %b)
	br :Finished

	:Equal
	call @printResult(%a, %b)

	:Finished
	return 
}

define @printResult(%a, %b) {
	%a <- %a << 1
	%a <- %a + 1
	call print(%a)

	%b <- %b << 1
	%b <- %b + 1
	call print(%b)

	return
}
