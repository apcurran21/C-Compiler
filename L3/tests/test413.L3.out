3
3
30
20
10
20
10
10
10
10
75
5
70
5
65
5
60
5
55
5
50
5
45
5
40
5
35
5
30
5
25
5
20
5
15
5
10
5
5
5
5
5
